src/libraries/System.Net.Mail/tests/Unit/MailAddressTests/MailAddressDisplayNameTest.cs

```diff
@@ -21,6 +22,45 @@ public void MailAddress_WithUnicodeDisplayAndMailAddress_ToStringShouldReturnDis
             Assert.Equal(string.Format("\"{0}\" <{1}>", DisplayNameWithUnicode, Address), _mailAddress.ToString());
         }
 
+        [Theory]
+        [InlineData(Address, "test\"Display\"Name")]
+        [InlineData(Address, "Hello \"world hello\" world")]
+        [InlineData(Address, "Hello \"world")]
+        [InlineData(Address, "Hello \"\"world")]
+        [InlineData(Address, "\"")]
+        [InlineData(Address, "Hello \\\"world hello\\\" world")]
+        public void MailAddress_WithDoubleQuotesDisplayAndMailAddress_ToStringShouldReturnDisplayNameEscapeSequenceAndAddressInAngleBrackets(string address, string displayNameWithDoubleQuotes)
+        {
+            MailAddress mailAddress = new MailAddress(address, displayNameWithDoubleQuotes);
+            Assert.Equal(displayNameWithDoubleQuotes, mailAddress.DisplayName);
+            Assert.Equal(string.Format("\"{0}\" <{1}>", displayNameWithDoubleQuotes.Replace("\"", "\\\""), Address), mailAddress.ToString());
+        }
+
+        [Theory]
+        [InlineData(Address, "\"John Doe\"")]
+        [InlineData(Address, "\"\"")]
+        public void MailAddress_WithDoubleQuotesInBeginningAndInEndingDisplayAndMailAddress_ToStringShouldReturnDisplayNameEscapeSequenceAndAddressInAngleBrackets(string address, string displayNameWithDoubleQuotesInBeginningAndInEnding)
+        {
+            if (displayNameWithDoubleQuotesInBeginningAndInEnding.Length >= 2 && displayNameWithDoubleQuotesInBeginningAndInEnding[0] == '\"' && displayNameWithDoubleQuotesInBeginningAndInEnding[^1] == '\"')
+            {
+
+                displayNameWithDoubleQuotesInBeginningAndInEnding = displayNameWithDoubleQuotesInBeginningAndInEnding.Substring(1, displayNameWithDoubleQuotesInBeginningAndInEnding.Length - 2);
```

Comment:
This is not testing what we want though. You're stripping the quotes before you pass it to the `MailAddress` ctor.

Reply:
then what is suggested action here ?

Reply:
Move the substring after the call of the `MailAddress` ctor.